nibabel/cifti2/cifti2.py

```diff
@@ -1209,6 +1210,36 @@ def _to_xml_element(self):
             mat.append(mim._to_xml_element())
         return mat
 
+    def get_axis(self, index):
+        '''
+        Generates the Cifti2 axis for a given dimension
+
+        Parameters
+        ----------
+        index : int
+            Dimension for which we want to obtain the mapping.
+
+        Returns
+        -------
+        axis : :class:`.cifti2_axes.Axis`
+        '''
+        from . import cifti2_axes
+        return cifti2_axes.from_index_mapping(self.get_index_map(index))
+
+    def get_data_shape(self):
+        """
+        Returns data shape expected based on the CIFTI-2 header
+        """
+        from . import cifti2_axes
+        if len(self.mapped_indices) == 0:
+            return ()
+        base_shape = [-1 for _ in range(max(self.mapped_indices) + 1)]
```

Comment:
Are all `-1`s guaranteed to be replaced?

Also:

```suggestion
        base_shape = [-1] * (max(self.mapped_indices) + 1)
```

Reply:
No, the CIFTI-2 header could leave certain dimensions undefined. Similarly to headers which define the wrong array shape, this should not happen. The reason I set this to -1 is that this would automatically cause the shape check to fail (leading to a warning when loading or creating such a file and an error when trying to write it). 

Now that I think about this again, I should have set it to np.nan (which should be a bit more self-expanatory) and document this behaviour in the docstring. 

Alternatively, we could raise an error in the method if there are undefined dimensions. After all there is no valid shape. However, that would prevent people from loading files for which not all dimensions have been defined in the CIFTI-2 header.